### PR TITLE
Modernizes Admin Only Wizard Event, Department Revolt

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Department Revolt is a seriously old, broken piece a' buttah. I made it better.

### New additions:
Departments that are independent will have new members join as separatists for their department
Security is now a choosable department
Admins triggering the event can pick which department becomes independent, and choose whether to announce the new state
Nations are by default aggressive, and will try to start shit with other nations if there are more than one. (admins can disable, peaceful nations never declare war)
Nations have fluff objectives

A bit better flavortext for explaining what the fuck is going on
### Fixes:
The event will no longer pick departments that are already independent
The event will no longer create empty teams if it chooses a department with no valid members (0 assistants)
The event can no longer create multiple of the same department's independence for different members...? Weird stuff folks

## Why It's Good For The Game

This admin button sucked

## Changelog
:cl:
admin: Department Revolt event is now... at least usable.
add: Department Revolt now hits security
/:cl:

### CODE REVIEW

File: code/modules/antagonists/separatist/separatist.dm [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+	potential_recruits = _potential_recruits
+
+/datum/team/nation/Destroy(force, ...)
+	UnregisterSignal(SSdcs, COMSIG_GLOB_CREWMEMBER_JOINED)
+	. = ..()
+
+/datum/team/nation/proc/new_possible_separatist(mob/living/crewmember, rank)
+	if(rank in potential_recruits)
+		//surely we can trust the player who just joined the game to have a mind.
+		living.mind.add_antag_datum(/datum/antagonist/separatist,src)
```
Comment (Fikou) [inline code/modules/antagonists/separatist/separatist.dm:19]:
```suggestion
		crewmember.mind.add_antag_datum(/datum/antagonist/separatist,src)
```

Comment (tralezab) [reply code/modules/antagonists/separatist/separatist.dm:19]: Thank you

**Comment by ATH1909** [comment]
what if you added it back into summon events rotation 😳

**Comment by SkeletalElite** [comment]
> what if you added it back into summon events rotation 😳

If this was removed when it was possible to just get summon events x5, I see why, but now that you can only purchase summon events once why not add it back

**Comment by EgorDinamit** [comment]
Why did you make a "Hats - Copy.dmi"? Same with changes in .dme, seems random.

**Comment by tralezab** [comment]
@EgorDinamit mistake? possibly a mistake.

File: code/modules/antagonists/separatist/separatist.dm [hunk 2]
```diff
@@ -1,12 +1,93 @@
...
+	destroy.team = src
+	objectives += destroy
+	update_all_member_objectives("<span class='danger'>The nation of [attacking_nation] has declared the intent to conquer [src]! You have new objectives.</span>")
+
+/datum/team/nation/proc/update_all_member_objectives(message)
+	for(var/datum/mind/M in members)
+		var/datum/antagonist/separatist/needs_objectives = M.has_antag_datum(/datum/antagonist/separatist)
+		needs_objectives.objectives |= objectives
+		if(message)
+			to_chat(M.current, message)
```
Comment (Jared-Fogle) [inline code/modules/antagonists/separatist/separatist.dm:68]:
```suggestion
	for(var/datum/mind/member in members)
		var/datum/antagonist/separatist/needs_objectives = member.has_antag_datum(/datum/antagonist/separatist)
		needs_objectives.objectives |= objectives
		if(message)
			to_chat(member.current, message)
```

File: code/modules/antagonists/separatist/separatist.dm [hunk 3]
```diff
@@ -16,4 +97,56 @@
...
+	. = ..()
+	target_team = target_department
+	update_explanation_text()
+
+/datum/objective/destroy_nation/check_completion()
+	if(!target_team)
+		return TRUE
+
+	for(var/datum/antagonist/separatist/separatist_datum in GLOB.antagonists)
+		if(!(separatist_datum.nation.nation_department != target_team.nation_department)) //a separatist, but not one part of the department we need to destroy
```
Comment (Jared-Fogle) [inline code/modules/antagonists/separatist/separatist.dm:127]: Why are you inverting an `!=`? This is hard to read.

File: code/modules/events/wizard/departmentrevolt.dm [hunk 4]
```diff
@@ -1,56 +1,116 @@
...
+		nation_name = "The [prefix] of [nation_name]"
 
-	var/datum/team/nation/nation = new
+	var/datum/team/nation/nation = new(null, jobs_to_revolt, department)
 	nation.name = nation_name
+	var/datum/team/department_target //dodges unfortunate runtime
+	if(!independent_departments.len)
+		department_target = null
+	else
+		department_target = pick(independent_departments)
```
Comment (Jared-Fogle) [inline code/modules/events/wizard/departmentrevolt.dm:91]: Inverse this. You should not be doing `if...else` with the original condition being a negative.

File: code/modules/events/wizard/departmentrevolt.dm [hunk 5]
```diff
@@ -1,56 +1,116 @@
...
+	nation.generate_nation_objectives(dangerous, department_target)
 
 	for(var/i in GLOB.human_list)
 		var/mob/living/carbon/human/H = i
 		if(H.mind)
 			var/datum/mind/M = H.mind
-			if(M.assigned_role && !(M.has_antag_datum(/datum/antagonist)))
+			if(M.assigned_role)
```
Comment (Jared-Fogle) [inline code/modules/events/wizard/departmentrevolt.dm:98]: Restructure this and the `H.mind` check to early continue if they fail.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Also, it looks like you still have the Copy.dmi

**Comment by Jared-Fogle** [comment]
Remove the bundle file, you don't need it anymore. It's safer here.

File: code/modules/events/wizard/departmentrevolt.dm [hunk 6]
```diff
@@ -1,56 +1,116 @@
...
-			var/datum/mind/M = H.mind
-			if(M.assigned_role && !(M.has_antag_datum(/datum/antagonist)))
-				for(var/job in jobs_to_revolt)
-					if(M.assigned_role == job)
-						citizens += H
-						M.add_antag_datum(/datum/antagonist/separatist,nation)
-						H.log_message("Was made into a separatist, long live [nation_name]!", LOG_ATTACK, color="red")
+		if(!H.mind)
+			continue
+		var/datum/mind/M = H.mind
```
Comment (Jared-Fogle) [inline code/modules/events/wizard/departmentrevolt.dm:96]: Better variable name.

**Pull request merged**